test/e2e/network/dual_stack.go

```diff
@@ -435,6 +437,157 @@ var _ = SIGDescribe("[Feature:IPv6DualStackAlphaFeature] [LinuxOnly]", func() {
 	})
 	// TODO (khenidak add slice validation logic, since endpoint controller only operates
 	// on primary ClusterIP
+
+	// Service Granular Checks as in k8s.io/kubernetes/test/e2e/network/networking.go
+	// but using the secondary IP, so we run the same tests for each ClusterIP family
+	// TODO: enable/implement NodeIP checks once dual-stack addresses on Nodes are implemented
+	// xref: https://github.com/kubernetes/enhancements/pull/1665
+	ginkgo.Describe("Granular Checks: Services Secondary IP Family", func() {
+
+		ginkgo.It("should function for pod-Service: http", func() {
+			config := e2enetwork.NewNetworkingTestConfig(f, e2enetwork.EnableDualStack)
+			ginkgo.By(fmt.Sprintf("dialing(http) %v --> %v:%v (config.clusterIP)", config.TestContainerPod.Name, config.SecondaryClusterIP, e2enetwork.ClusterHTTPPort))
+			config.DialFromTestContainer("http", config.SecondaryClusterIP, e2enetwork.ClusterHTTPPort, config.MaxTries, 0, config.EndpointHostnames())
+
+			// Skip until we support Secondary Node IP
+			// ginkgo.By(fmt.Sprintf("dialing(http) %v --> %v:%v (nodeIP)", config.TestContainerPod.Name, config.SecondaryNodeIP, config.NodeHTTPPort))
+			// config.DialFromTestContainer("http", config.SecondaryNodeIP, config.NodeHTTPPort, config.MaxTries, 0, config.EndpointHostnames())
```

Comment:
We can run this test if the `SecondaryNodeIP` exists?

Reply:
I don't like "smart" tests :smile: 
Let me explain, when we run a test we want to verify something, that the code does what is supposed to do.
If we only run if not empty, we can't know if it's working or not, we know that the test passed , but we can not guarantee that SecondaryNodeIPs work.
Another option is to move it to a different test and skip, that is different than skipping on the test directly, because the report mentions that the test was skipped.
Anyway,  you are right https://github.com/kubernetes/kubernetes/pull/96484#discussion_r521697789 and this works now, I have to modify the test